### PR TITLE
[Documentation][Book] Payments - custom gateways

### LINKED ISSUE #7654: FOS\RestBundle exception on payment capture
Git dev-master commit: b01c34a3fc1a835404ff849e6ebf6d725797f6d6

After adding a custom gateway in the admin panel, on payment capture I get an error when using this gateway. It does not look like it even gets to my custom payment bundle and errors out way earlier

```
FatalThrowableError in ExceptionController.php line 43:
Type error: Argument 1 passed to FOS\RestBundle\Controller\ExceptionController::__construct() must be an instance of FOS\RestBundle\View\ViewHandlerInterface, none given, called in /var/www/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php on line 202
in ExceptionController.php line 43
at ExceptionController->__construct() in ControllerResolver.php line 202
at ControllerResolver->instantiateController('FOS\\RestBundle\\Controller\\ExceptionController') in ControllerResolver.php line 85
at ControllerResolver->instantiateController('FOS\\RestBundle\\Controller\\ExceptionController') in ControllerResolver.php line 190
at ControllerResolver->createController('FOS\\RestBundle\\Controller\\ExceptionController::showAction') in ControllerResolver.php line 73
at ControllerResolver->createController('FOS\\RestBundle\\Controller\\ExceptionController::showAction') in ControllerResolver.php line 95
at ControllerResolver->getController(object(Request)) in TraceableControllerResolver.php line 58
at TraceableControllerResolver->getController(object(Request)) in HttpKernel.php line 136
at HttpKernel->handleRaw(object(Request), 2) in HttpKernel.php line 68
at HttpKernel->handle(object(Request), 2, false) in ExceptionListener.php line 50
at ExceptionListener->onKernelException(object(GetResponseForExceptionEvent), 'kernel.exception', object(TraceableEventDispatcher))
at call_user_func(array(object(ExceptionListener), 'onKernelException'), object(GetResponseForExceptionEvent), 'kernel.exception', object(TraceableEventDispatcher)) in WrappedListener.php line 106
at WrappedListener->__invoke(object(GetResponseForExceptionEvent), 'kernel.exception', object(ContainerAwareEventDispatcher))
at call_user_func(object(WrappedListener), object(GetResponseForExceptionEvent), 'kernel.exception', object(ContainerAwareEventDispatcher)) in EventDispatcher.php line 174
at EventDispatcher->doDispatch(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.exception', object(GetResponseForExceptionEvent)) in EventDispatcher.php line 43
at EventDispatcher->dispatch('kernel.exception', object(GetResponseForExceptionEvent)) in TraceableEventDispatcher.php line 136
at TraceableEventDispatcher->dispatch('kernel.exception', object(GetResponseForExceptionEvent)) in HttpKernel.php line 230
at HttpKernel->handleException(object(InvalidArgumentException), object(Request), 1) in HttpKernel.php line 79
at HttpKernel->handle(object(Request), 1, true) in Kernel.php line 168
at Kernel->handle(object(Request)) in app_dev.php line 43
```

Form Type for admin panel looks like this:
```php
<?php

namespace TransactProBundle\Form;

use Symfony\Component\Form\AbstractType;
use Symfony\Component\Form\Extension\Core\Type\TextType;
use Symfony\Component\Form\FormBuilderInterface;
use Symfony\Component\Validator\Constraints\NotBlank;
use \Symfony\Component\Form\Extension\Core\Type\ChoiceType;

class TransactProGatewayConfigurationType extends AbstractType
{
    /**
     * {@inheritdoc}
     */
    public function buildForm(FormBuilderInterface $builder, array $options)
    {
        $builder
            ->add('guid', TextType::class, [
                'label' => 'sylius.form.gateway_configuration.transactpro.guid',
                'constraints' => [
                    new NotBlank([
                        'message' => 'sylius.gateway_config.transactpro.guid.not_blank',
                        'groups' => 'sylius',
                    ])
                ],
            ])
            ->add('password', TextType::class, [
                'label' => 'sylius.form.gateway_configuration.transactpro.password',
                'constraints' => [
                    new NotBlank([
                        'message' => 'sylius.gateway_config.transactpro.password.not_blank',
                        'groups' => 'sylius',
                    ])
                ],
            ])
            ->add('routing_string', TextType::class, [
                'label' => 'sylius.form.gateway_configuration.transactpro.routing_string',
                'constraints' => [
                    new NotBlank([
                        'message' => 'sylius.gateway_config.transactpro.routing_string.not_blank',
                        'groups' => 'sylius',
                    ])
                ],
            ])
            ->add('sandbox', ChoiceType::class, [
                'label' => 'sylius.form.gateway_configuration.transactpro.sandbox',
                'choices' => ['yes' => true, 'no' => false]
            ])
        ;
    }
}
```

Service registration
```yml
services:
    sylius.form.type.gateway_configuration.transactpro:
        class: TransactProBundle\Form\TransactProGatewayConfigurationType
        tags:
            - { name: sylius.gateway_configuration_type, type: transactpro_checkout, label: sylius.payum_gateway_factory.transactpro_checkout }
            - { name: form.type }
```

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | yes |
| New docs?    | no |
| Related tickets | fixes #7654 |
| License         | MIT |

### CODE REVIEW

File: docs/book/orders/payments.rst [hunk 1]
```diff
@@ -99,12 +99,41 @@ In order to have your new payment method available in the checkout remember to *
...
 
-In order to add a new gateway, you should register it's configuration form type with ``sylius.gateway_configuration_type`` tag. Then it will be available in the Admin panel in the gateway choice dropdown.
+Payment Gateways that already have a Sylius bridge
+''''''''''''''''''''''''''''''''''''''''''''''''''
+
+First you need to create the configuration form type for your gateway. Have a look at the configuration form types of
+`Paypal <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/PayumBundle/Form/Type/PaypalGatewayConfigurationType.php>`_
+and `Stripe <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/PayumBundle/Form/Type/StripeGatewayConfigurationType.php>`_.
+
+Then you should register it's configuration form type with ``sylius.gateway_configuration_type`` tag.
```
Comment (venyii) [inline docs/book/orders/payments.rst]: it's -> its

File: docs/book/orders/payments.rst [hunk 2]
```diff
@@ -99,12 +99,41 @@ In order to have your new payment method available in the checkout remember to *
...
 
+Other Payment Gateways
+''''''''''''''''''''''
+
+.. note::
+
+    Learn more about integrating payment gateways in `the Payum docs <https://github.com/Payum/Payum/blob/master/docs/index.md>`_.
+
+When the Payment Gateway you are trying to use does have a bridge available and you integrate them on your own,
+use our guide on :doc:`extensions development </cookbook/creating-extension>`.
```
Comment (venyii) [inline docs/book/orders/payments.rst]: extension~~s~~ development ?

Comment (TheMadeleine) [inline docs/book/orders/payments.rst]: why not plural? :)

Comment (venyii) [inline docs/book/orders/payments.rst]: Hmm sounded weird to me, it also links to creating an extension, not creating extensions.

But fine for me 🤷‍♂️

**Comment by pjedrzejewski** [comment]
Thank you Magda!

**Pull request merged**